python/lsst/daf/butler_migrate/database.py

```diff
@@ -51,6 +53,12 @@ class Database:
         Database schema/namespace.
     """
 
+    dimensions_json_key = "config:dimensions.json"
```

Comment:
Indeed there is some commonality in these two classes. They are separate because they work in different contexts, `ButlerAttributes` is used in migration context where connection is established by alembic, and `Database` is for use by CLI methods before things get passed to alembic. I probably need to refactor it to avoid duplication, I just don't know yet how to do it.